### PR TITLE
Add auto_length parameter and enum length checking to EnumField

### PR DESCRIPTION
**Pull request by ADR-007**
PR with adding the auto_legnth parameter to the EnumField and checking that the enum values fit into the max_length of the field.

### CODE REVIEW

File: enumfields/fields.py [hunk 1]
```diff
@@ -128,11 +128,24 @@ def formfield(self, form_class=None, choices_form_class=None, **kwargs):
 
 
 class EnumField(EnumFieldMixin, models.CharField):
-    def __init__(self, enum, **kwargs):
-        kwargs.setdefault("max_length", 10)
+    def __init__(self, enum, auto_length=False, **kwargs):
+        max_item_length = self._get_max_item_length(enum)
+        kwargs.setdefault("max_length", (max_item_length // 10 + 1) * 10 if auto_length else 10)
+
+        assert max_item_length <= kwargs["max_length"], 'The enum have value longer that max_length of the field!'
```
Comment (akx) [inline enumfields/fields.py]: This shouldn't be an `assert` but a real `if`/`raise ValueError`. (`assert`s could be optimized away.)

Also, maybe "Enum has value(s) longer than the declared max_length (%d)" or similar?

File: enumfields/fields.py [hunk 2]
```diff
@@ -128,11 +128,24 @@ def formfield(self, form_class=None, choices_form_class=None, **kwargs):
...
+
         super(EnumField, self).__init__(enum, **kwargs)
         self.validators = []
 
+    @staticmethod
+    def _get_max_item_length(enum):
```
Comment (akx) [inline enumfields/fields.py]: There's no reason for this to be a class staticmethod; just make it a free function instead?

File: enumfields/fields.py (hunk 1 above)
Comment (ADR-007) [reply enumfields/fields.py]: About `assert`: I think we have to check this during developing and CI tests only but not in the production. It is the way that used by Django REST Framework. (But `raise ValueError` is also OK, so I can change it.)
What do you think?

About message: agree.

File: enumfields/fields.py (hunk 2 above)
Comment (ADR-007) [reply enumfields/fields.py]: Can I add such method to `Enum` class?

**Pull request closed without merging**